packages/discovery-provider/src/api/v1/tracks.py

```diff
@@ -546,6 +531,121 @@ def get(self, track_id):
+        description="""Download an original or mp3 track""",
+        params={"track_id": "A Track ID"},
+        responses={
+            200: "Success",
+            216: "Partial content",
+            400: "Bad request",
+            416: "Content range invalid",
+            500: "Server error",
+        },
+    )
+    @ns.expect(download_parser)
+    @cache(ttl_sec=5, transform=tranform_stream_cache)
+    def get(self, track_id):
+        """
+        Download the original or MP3 file of a track.
+        """
+        request_args = download_parser.parse_args()
+        decoded_id = decode_with_abort(track_id, ns)
+        info = get_track_access_info(decoded_id)
+        track = info.get("track")
+        if not track:
+            logger.error(
+                f"tracks.py | download | Track with id {track_id} may not exist. Please investigate."
+            )
+            abort_not_found(track_id, ns)
+
+        redis = redis_connection.get_redis()
+
+        # signature for the track to be included as a query param in the redirect to CN
+        download_signature = get_track_download_signature(
+            {
+                "track": track,
+                "is_original": request_args.get("original"),
+                "filename": request_args.get("filename"),
+                "user_data": request_args.get("user_data"),
+                "user_signature": request_args.get("user_signature"),
+            }
+        )
+        if not download_signature:
+            abort_not_found(track_id, ns)
+
+        signature = download_signature["signature"]
+        cid = download_signature["cid"]
+        filename = download_signature["filename"]
+        params = {"signature": json.dumps(signature), "filename": filename}
+        base_path = f"tracks/cidstream/{cid}"
+        query_string = urllib.parse.urlencode(params, quote_via=urllib.parse.quote)
+        path = f"{base_path}?{query_string}"
+
+        # we cache track cid -> content node so we can avoid
+        # checking multiple content nodes for a track
+        # if we already know where to look
+        redis_key = f"track_cid:{cid}"
+        cached_content_node = redis.get(redis_key)
+        stream_url = None
+        if cached_content_node:
+            cached_content_node = cached_content_node.decode("utf-8")
+            stream_url = get_stream_url_from_content_node(cached_content_node, path)
+            if stream_url:
+                return stream_url
+
+        healthy_nodes = get_all_healthy_content_nodes_cached(redis)
+        if not healthy_nodes:
+            logger.error(
+                f"tracks.py | stream | No healthy Content Nodes found when streaming track ID {track_id}. Please investigate."
+            )
+            abort_not_found(track_id, ns)
+
+        rendezvous = RendezvousHash(
+            *[re.sub("/$", "", node["endpoint"].lower()) for node in healthy_nodes]
+        )
+
+        content_nodes = rendezvous.get_n(9999999, cid)
```

Comment:
I know you copied this from stream, but it's interesting - do we really need that many 👀

Reply:
which part are you talking about exactly?

Reply:
9999999

Reply:
probably not, but going to keep as is for consistency.